src/main/java/com/google/sps/servlets/ReviewsServlet.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import static com.google.sps.Constants.*;
+import com.google.sps.Review;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Servlet that handles customer reviews.
+ */
+@WebServlet("/reviews")
+public class ReviewsServlet extends HttpServlet {
+  Gson gson;
+  DatastoreService datastore;
+
+  public ReviewsServlet() throws IOException {
+    gson = new Gson();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  /** Get review from review form and store it in Datastore under its Business entity. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get review input from review form.
+    String userID = getParameter(request, "name", "anonymous"); // Accept input name for now, should be user Datastore key, to be implemented later
+    String comment = getParameter(request, "review", "404");
```

Comment:
the 404 standard should just be for floats or numbers. In your case you might wanna use empty string instead, or "No description provided"?

Reply:
Ah yes, edited!